src/types.c

```diff
@@ -133,6 +136,11 @@ bool is_number_type(const Type *t)
 {
     return is_integer_type(t) || t->kind == TYPE_DOUBLE;
 }
+
+bool is_float_type(const Type *t)
+{
+    return is_integer_type(t) || t->kind == TYPE_FLOAT;

```

Comment:
This still doesn't make any sense to me. Why are types such as `int` considered -"float type" here, and why is `double` not a "float type"?

Reply:
so, I need remove the is_float_type function?

Reply:
Yes, we can remove it for now.

Reply:
okay, fixed, and repalced ```is_float_type``` to ```is_number_type```